### PR TITLE
Identity,Jwt Token

### PR DESCRIPTION
**Pull request by PlacidaRebello**

### CODE REVIEW

File: BussinessLogic/RegisterLogic.cs [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+        private UserManager<IdentityUser> _userManager { get; }
+        public RegisterLogic(UserManager<IdentityUser> userManager)
+        {
+            _userManager = userManager;
+        }
+
+        public async Task<bool> RegisterUser(AppUser user, string password)
+        {
+            //bool result= await  _registerEngine.RegisterUser(identityUser, password);
+            // return result;
```
Comment (diasjason) [inline BussinessLogic/RegisterLogic.cs]: cleanup

File: DataAccess/DataAccess.csproj [hunk 2]
```diff
@@ -5,12 +5,18 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="Microsoft.EntityFrameworkCore.SqlServer" Version="3.0.1" />
-    <PackageReference Include="Microsoft.EntityFrameworkCore.Tools" Version="3.0.1">
+    <Compile Remove="Interfaces\IRegisterEngine.cs" />
+    <Compile Remove="Models\User.cs" />
+    <Compile Remove="RegisterEngine.cs" />
```
Comment (diasjason) [inline DataAccess/DataAccess.csproj]: this is happening coz you excluded some files. include them and delete them

**Comment by PlacidaRebello** [comment]
How do I include them now...I can't c them in the project now

File: DataAccess/RegisterEngine.cs [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+namespace DataAccess
+{
+    public class RegisterEngine : IRegisterEngine
+    {
+        public RegisterEngine(UserManager<IdentityUser> userManager)
+        {
+            UserManager = userManager;
+        }
+
+        public UserManager<IdentityUser> UserManager { get; }
```
Comment (diasjason) [inline DataAccess/RegisterEngine.cs]: this should be private and named _userManager

**Comment by diasjason** [comment]
you can try rightclick on the parent folder and add existing items

File: IssueTracker/Controllers/RegisterController.cs [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+using DataAccess.Models;
+
+namespace IssueTracker.Controllers
+{
+    [Route("api/[controller]")]
+    [ApiController]
+    public class RegisterController : ControllerBase
+    {
+        public UserManager<IdentityUser> UserManager { get; }
+        private SignInManager<IdentityUser> _signInManager { get; }
```
Comment (diasjason) [inline IssueTracker/Controllers/RegisterController.cs]: make these private and readonly for consistency

File: IssueTracker/Controllers/RegisterController.cs [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+        private readonly IRegisterLogic _registerLogic;
+        private readonly IMapper _mapper;
+        public RegisterController(UserManager<IdentityUser> userManager, SignInManager<IdentityUser> signInManager, IMapper mapper, IRegisterLogic registerLogic)
+        {
+            this.UserManager = userManager;
```
Comment (diasjason) [inline IssueTracker/Controllers/RegisterController.cs]: dont need this.

File: IssueTracker/Controllers/RegisterController.cs [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+            _signInManager = signInManager;
+            _mapper = mapper;
+            _registerLogic = registerLogic;
+        }
+
+        [HttpPost]
+        public async Task<CreateResponse> CreateUserAsync(RegisterUserRequest userRequest)
+        {
+            var newuser = _mapper.Map<AppUser>(userRequest);
```
Comment (diasjason) [inline IssueTracker/Controllers/RegisterController.cs]: param should be named either user or newUser

File: IssueTracker/Controllers/SignInController.cs [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+using ServiceModel.Type;
+
+namespace IssueTracker.Controllers
+{
+    [Route("api/[controller]")]
+    [ApiController]
+    public class SignInController : ControllerBase
+    {
+        private UserManager<IdentityUser> _userManager { get; }
+        private SignInManager<IdentityUser> _signInManager { get; }
```
Comment (diasjason) [inline IssueTracker/Controllers/SignInController.cs:23]: readonly here as well

File: IssueTracker/Controllers/SignInController.cs [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+            }
+            return Unauthorized();
+        }
+
+        private object Createtoken(string userName)
+        {
+            var authClaims = new[]
+            {
+                    new Claim(JwtRegisteredClaimNames.Sub, userName),
+                    new Claim(JwtRegisteredClaimNames.Jti, Guid.NewGuid().ToString())
```
Comment (diasjason) [inline IssueTracker/Controllers/SignInController.cs]: remove unnecessary tabs

File: IssueTracker/Controllers/SignInController.cs [hunk 9]
```diff
@@ -0,0 +1,66 @@
...
+                issuer: _authOptions.Issuer,
+                audience: _authOptions.Audience,
+                expires: DateTime.Now.AddHours(_authOptions.ExpiresInMinutes),
+                claims: authClaims,
+                signingCredentials: new SigningCredentials(new SymmetricSecurityKey(Encoding.UTF8.GetBytes(_authOptions.SecureKey)),
+                    SecurityAlgorithms.HmacSha256Signature)
+                );
+            return Ok(new
+            {
+                token = "Bearer " + new JwtSecurityTokenHandler().WriteToken(token),
```
Comment (diasjason) [inline IssueTracker/Controllers/SignInController.cs]: use string interpolation. i know its all my code, but I warned you not to use it as it is :D

File: IssueTracker/Controllers/StatusController.cs [hunk 10]
```diff
@@ -19,70 +19,60 @@ public class StatusController : ControllerBase
...
-
             _statusLogic.EditStatus(newStatus);
-
-            return new CreateStatusResponse
-            { StatusId = newStatus.StatusId,
-              Message="Edited"
+            return new CreateResponse
+            {
+                Id = newStatus.StatusId,
+                Message = "Edited successfully"
```
Comment (diasjason) [inline IssueTracker/Controllers/StatusController.cs]: use capital S in successfully for consistency

File: IssueTracker/IssueTracker.csproj.user [hunk 11]
```diff
@@ -10,5 +10,6 @@
     <WebStackScaffolding_LayoutPageFile />
     <WebStackScaffolding_DbContextTypeFullName>DataAccess.Models.DataContext</WebStackScaffolding_DbContextTypeFullName>
     <WebStackScaffolding_IsAsyncSelected>False</WebStackScaffolding_IsAsyncSelected>
+    <ShowAllFiles>false</ShowAllFiles>
```
Comment (diasjason) [inline IssueTracker/IssueTracker.csproj.user:13]: this is caused by hiding files too. remove this and see if excluded files show up

File: IssueTracker/Startup.cs [hunk 12]
```diff
@@ -40,18 +49,73 @@ public void ConfigureServices(IServiceCollection services)
             services.AddTransient<IIssueTypeLogic, IssueTypeLogic>();
             services.AddTransient<IIssueTypeEngine, IssueTypeEngine>();
 
+            services.AddTransient<IRegisterLogic, RegisterLogic>();
+            //  services.AddTransient<IRegisterEngine, RegisterEngine>();
```
Comment (diasjason) [inline IssueTracker/Startup.cs]: remove comment

File: IssueTracker/Startup.cs [hunk 13]
```diff
@@ -40,18 +49,73 @@ public void ConfigureServices(IServiceCollection services)
...
+                    Description =
+                        "JWT Authorization header using the Bearer scheme. \r\n\r\n Enter 'Bearer' [space] and then your token in the text input below.\r\n\r\nExample: \"Bearer 12345abcdef\"",
+                    Name = "Authorization",
+                    In = ParameterLocation.Header,
+                    Type = SecuritySchemeType.ApiKey,
+                    Scheme = "Bearer"
+                });
+
+                c.AddSecurityRequirement(new OpenApiSecurityRequirement()
+               {
```
Comment (diasjason) [inline IssueTracker/Startup.cs]: fix indentation and spaces

File: ServiceModel/ServiceModel.csproj [hunk 14]
```diff
@@ -4,6 +4,12 @@
     <TargetFramework>netstandard2.1</TargetFramework>
   </PropertyGroup>
 
+  <ItemGroup>
+    <Compile Remove="Dto\CreateIssueResponse.cs" />
+    <Compile Remove="Dto\CreateIssueTypeResponse.cs" />
+    <Compile Remove="Dto\CreateStatusResponse.cs" />
+  </ItemGroup>
```
Comment (diasjason) [inline ServiceModel/ServiceModel.csproj]: this is also caused by excluding files

**Review by diasjason — CHANGES_REQUESTED** [review]

File: IssueTracker/appsettings.json [hunk 15]
```diff
@@ -7,9 +7,9 @@
     }
   },
   "ConnectionStrings": {
-    //"myconn": "server=DESKTOP-URSHK4B\\SQLEXPRESS; database=IssueTrackerDb;Trusted_Connection=True;"
+    "myconn": "server=DESKTOP-URSHK4B\\SQLEXPRESS; database=IssueTrackerDb;Trusted_Connection=True;"
```
Comment (diasjason) [inline IssueTracker/appsettings.json]: make sure you are publising to azure db and not local db

**Review by diasjason — CHANGES_REQUESTED** [review]

File: BussinessLogic/BussinessLogic.csproj [hunk 16]
```diff
@@ -5,7 +5,8 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="System.ComponentModel.Annotations" Version="4.6.0" />
+    <Compile Remove="Interfaces\IUserLogic.cs" />
+    <Compile Remove="UserLogic.cs" />
```
Comment (diasjason) [inline BussinessLogic/BussinessLogic.csproj]: remove this aswell

**Review by diasjason — CHANGES_REQUESTED** [review]

**Pull request merged**